src/taipy/core/pipeline/_pipeline_manager.py

```diff
@@ -9,42 +9,60 @@
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
 
+import json
+import pathlib
 from functools import partial
-from typing import Any, Callable, List, Optional, Union
-
-from taipy.config.common.scope import Scope
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
 
 from .._entity._entity_ids import _EntityIds
 from .._manager._manager import _Manager
-from .._repository._abstract_repository import _AbstractRepository
 from .._version._version_mixin import _VersionMixin
+from ..common._utils import _Subscriber
 from ..common.warn_if_inputs_not_ready import _warn_if_inputs_not_ready
-from ..config.pipeline_config import PipelineConfig
-from ..cycle.cycle_id import CycleId
-from ..exceptions.exceptions import NonExistingPipeline
+from ..exceptions.exceptions import (
+    InvalidPipelineId,
+    ModelNotFound,
+    NonExistingPipeline,
+    NonExistingTask,
+    PipelineBelongsToNonExistingScenario,
+)
 from ..job._job_manager_factory import _JobManagerFactory
 from ..job.job import Job
 from ..notification import EventEntityType, EventOperation, _publish_event
+from ..scenario._scenario_manager_factory import _ScenarioManagerFactory
+from ..scenario.scenario import Scenario
 from ..scenario.scenario_id import ScenarioId
 from ..task._task_manager_factory import _TaskManagerFactory
-from ..task.task import Task
+from ..task.task import Task, TaskId
 from .pipeline import Pipeline
 from .pipeline_id import PipelineId
 
 
 class _PipelineManager(_Manager[Pipeline], _VersionMixin):
 
     _ENTITY_NAME = Pipeline.__name__
-    _repository: _AbstractRepository
     _EVENT_ENTITY_TYPE = EventEntityType.PIPELINE
+    _dir_name = "pipelines"
 
     @classmethod
     def _get_all(cls, version_number: Optional[str] = None) -> List[Pipeline]:
         """
-        Returns all entities.
+        Returns all Pipeline entities.
         """
-        filters = cls._build_filters_with_version(version_number)
-        return cls._repository._load_all(filters)
+        pipelines = []
+        scenarios = _ScenarioManagerFactory._build_manager()._get_all(version_number)
+        for scenario in scenarios:
+            pipelines.extend(list(scenario.pipelines.values()))
+
+        return pipelines
+
+    @classmethod
+    def _get_all_by(cls, filters: Optional[List[Dict]] = None) -> List[Pipeline]:
```

Comment:
This method is weird imo.

We filter the scenario, then get all pipelines.
But what we should actually do is get all pipelines, then filter those pipelines.

However, this method did not exist before. Why do we need to add it?

Reply:
haha I understand your confusion as I did too. The logic behind this is because we're storing Pipeline in ScenarioModel now, so we will need to filter ScenarioModel for data related to Pipeline and that's the reason behind why we filter the scenario 1st. 
As for the 2nd point, yep I don't think it's used anywhere aside from the test. But just to be sure that we stay consistent with the _Manager class, I think it'd have more benefit to implement them also instead of removing it.

Reply:
If some pipelines have a custom attribute named `description` (which should not, but just assuming), and no scenario has any `description`, then `PipelineManager._get_all_by({"description": "foo"})` would return empty.

I don't think this is a safe behavior.

Reply:
Can we use the `PipelineManager._get_all()` method, and then filter from all the pipelines?

Reply:
we can but that'd hurt the performance actually 😅

Reply:
We are not using it, so the performance is not a problem atm.

And we need to make sure that if we do use it, the behavior is correct.

Reply:
hmmm, but we also have some share data of pipeline that is being stored in scenariomodel, so without filtering by the scenario, I'm not sure we can get the correct data either, like version for example is being shared with scenario. maybe the best solution is to filter by scenario first, then filter again by pipeline?

Reply:
I have updated it to the solution proposed by Giang, to get all pipelines and check by the filter provided with getattr :D